### PR TITLE
verify lsvpd information

### PR DESCRIPTION
**Pull request by Naresh-ibm**
check the lsvpd pci info with sys pci inf lsvpd information

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

File: io/pci/pci_info_lsvpd.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                vpd_device_id = pci.get_vpd(i)['pci_id'][4:]
+                sysfs_device_id = pci.get_pci_id_from_sysfs(i)[5:-10]
+                if vpd_device_id == sysfs_device_id:
+                    print "Device ID from sysfs :", sysfs_device_id
+                    print "Device ID from vpd:", vpd_device_id
+                    print "=======>>Match Success\n\n"
+                else:
+                    print "Device ID from sysfs :", sysfs_device_id
+                    print "Device ID from vpd:", vpd_device_id
+                    print "----->>Device ID did not Match\n\n"
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]:
```
print "Device ID from sysfs :", sysfs_device_id
print "Device ID from vpd:", vpd_device_id
```
can be done only once, maybe before comparison

Comment (PraveenPenguin) [inline io/pci/pci_info_lsvpd.py]: print statement not looks good can you please use self.log.info(message) ?

File: io/pci/pci_info_lsvpd.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        '''
+        sm = SoftwareManager()
+        for pkg in ["lsvpd"]:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+
+    def test_pcilsvpdinfo(self):
+        error = []
+        for i in pci.get_pci_addresses():
+            print "Comparing for PCI Domain  : ", i, "\n\n"
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: `Comparing for PCI Address`

File: io/pci/pci_info_lsvpd.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            if pci.get_vpd(i):
+                '''
+                Slot Match
+                '''
+                if pci.get_vpd(i).has_key('slot'):
+                    sys_slot = pci.get_slot_from_sysfs(i).strip('\0')
+                    vpd_slot = pci.get_vpd(i)['slot']
+                    vpd1_slot = vpd_slot[:vpd_slot.rfind('-')]
+                    if vpd_slot == sys_slot or sys_slot == vpd1_slot:
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: `if sys_slot  == vpd_slot or sys_slot == vpd1_slot:`

File: io/pci/pci_info_lsvpd.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        for i in pci.get_pci_addresses():
+            print "Comparing for PCI Domain  : ", i, "\n\n"
+            if pci.get_vpd(i):
+                '''
+                Slot Match
+                '''
+                if pci.get_vpd(i).has_key('slot'):
+                    sys_slot = pci.get_slot_from_sysfs(i).strip('\0')
+                    vpd_slot = pci.get_vpd(i)['slot']
+                    vpd1_slot = vpd_slot[:vpd_slot.rfind('-')]
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: change `vpd1_slot ` to `vpd_slot_stripped` or something understandible

File: io/pci/pci_info_lsvpd.py (hunk 1 above)
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: above changes in all places applicable

File: io/pci/pci_info_lsvpd.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                '''
+                if pci.get_pci_id_from_sysfs(i) == pci.get_pci_id(i):
+                    print "from Sysfs :", pci.get_pci_id_from_sysfs(i)
+                    print "from Vpd :", pci.get_pci_id(i)
+                    print "========>>>> All PCI ID match Success\n\n"
+                else:
+                    print "---->>>PCI info Did not Matches\n\n"
+                    error.append(i)
+
+        print "Error PCI Domains"
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]:
```
if error:
    print "Error PCI Domains"
    print error
```

**Comment by narasimhan-v** [comment]
@Naresh-ibm attach job.log

**Comment by narasimhan-v** [comment]
@Naresh-ibm create a README file for the test

File: io/pci/pci_info_lsvpd.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Naresh Bannoth <nbannoth@linux.vnet.ibm.com>
+
+"""
+PCI device Test.
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: PCI VPD Test

File: io/pci/pci_info_lsvpd.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        for pkg in ["lsvpd"]:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+
+    def test_pcilsvpdinfo(self):
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: Docstring missing

File: io/pci/pci_info_lsvpd.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+                if pci.get_vpd(i).has_key('slot'):
+                    sys_slot = pci.get_slot_from_sysfs(i).strip('\0')
+                    vpd_slot = pci.get_vpd(i)['slot']
+                    vpd1_slot = vpd_slot[:vpd_slot.rfind('-')]
+                    if vpd_slot == sys_slot or sys_slot == vpd1_slot:
+                        print "Slot from sysfs:", sys_slot
+                        print "Slot from lsvpd:", pci.get_vpd(i)['slot']
+                        print "=======>>> slot matches perfectly\n\n"
+                    else:
+                        error.append(i)
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: Errors don't indicate which test failed
`errors.append(i + " slot ")` would be better

**Comment by narasimhan-v** [comment]
There are 4 sub tests. Can each become a function of its own

File: io/pci/pci_info_lsvpd.py [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        for pkg in ["lsvpd"]:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+
+    def test_pcilsvpdinfo(self):
+        error = []
+        for i in pci.get_pci_addresses():
```
Comment (PraveenPenguin) [inline io/pci/pci_info_lsvpd.py]: just want to add one more thing please use variable name as meaningful as `i`  not looks good

File: io/pci/pci_info_lsvpd.py (hunk 3 above)
Comment (PraveenPenguin) [inline io/pci/pci_info_lsvpd.py]: can it be if sys_slot in ['vpd_slot', 'vpd1_slot'] ?

**Comment by Naresh-ibm** [comment]
Thanks @narasimhan-v  and @PraveenPenguin  for all suggestions. learnt good amount of things.
made all the suggested changes. please let me know if anything is missing.
attaching a job.log file for your reference.

[pci_info_lsvpd_job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/755032/pci_info_lsvpd_job.txt)

**Comment by PraveenPenguin** [comment]
@Naresh-ibm can  you please squash the commits into one commit only ?

**Comment by narasimhan-v** [comment]
@Naresh-ibm You are running lsvpd for a particular pci address multiple times. You can run it once, store the results and use the same for all tests for that pci address

**Comment by narasimhan-v** [comment]
Also, correct the typo **info** in commit message

File: io/pci/pci_info_lsvpd.py [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+class PciLsvpdInfo(Test):
+    '''
+    check lsvpd info
+    '''
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: `smm` please

File: io/pci/pci_info_lsvpd.py [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+                    self.log.info("Slot from lsvpd:%s", vpd_slot)
+                    if sys_slot == vpd_slot or sys_slot == vpd_slot_stripped:
+                        self.log.info("=======>>> slot matches perfectly\n\n")
+                    else:
+                        error.append(pci_adres + "->slot")
+                        self.log.info("--->>Slot Numbers not Matched\n\n")
+                else:
+                    self.log.error("Slot info not available in vpd output\n")
+                '''
+                Device ID match"
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: " not removed

File: io/pci/pci_info_lsvpd.py [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+                    error.append(pci_adres + "->Subvendor_id")
+
+                '''
+                PCI ID Match
+                '''
+                sys_pci_id = pci.get_pci_id_from_sysfs(pci_adres)
+                self.log.info("from Sysfs:%s", sys_pci_id)
+                self.log.info("from Vpd :%s", pci.get_pci_id(pci_adres))
+
+                if sys_pci_id == pci.get_pci_id(pci_adres):
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: Running `get_pci_id` twice here

File: io/pci/pci_info_lsvpd.py [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+                    self.log.info("======>>>> All PCI ID match Success\n\n")
+                else:
+                    self.log.error("---->>>PCI info Did not Matches\n\n")
+                    error.append(pci_adres + "->pci_id")
+
+        if error:
+            self.log.error("failed domain:%s", error)
+            self.fail("Data mismatched for above Domains: ")
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: They are not domains, they are pci_addresses

File: io/pci/pci_info_lsvpd.py [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+        start Test execution
+        '''
+        error = []
+        for pci_adres in pci.get_pci_addresses():
+            print "Comparing for PCI Adress : ", pci_adres, "\n\n"
+            if pci.get_vpd(pci_adres):
+                '''
+                Slot Match
+                '''
+                if pci.get_vpd(pci_adres).has_key('slot'):
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: Running `pci.get_vpd()` multiple times.

File: io/pci/pci_info_lsvpd.py [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+                    sys_slot = pci.get_slot_from_sysfs(pci_adres).strip('\0')
+                    vpd_slot = pci.get_vpd(pci_adres)['slot']
+                    vpd_slot_stripped = vpd_slot[:vpd_slot.rfind('-')]
+                    self.log.info("Slot from sysfs:%s", sys_slot)
+                    self.log.info("Slot from lsvpd:%s", vpd_slot)
+                    if sys_slot == vpd_slot or sys_slot == vpd_slot_stripped:
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: `if sys_slot == vpd_slot or sys_slot == vpd_slot[:vpd_slot.rfind('-')]` would save execution time

File: io/pci/pci_info_lsvpd.py [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+                    self.log.info("======>>>Subvendor ID Match Success\n\n")
+                else:
+                    self.log.error("---->>Subvendor_id Not Matched\n\n")
+                    error.append(pci_adres + "->Subvendor_id")
+
+                '''
+                PCI ID Match
+                '''
+                sys_pci_id = pci.get_pci_id_from_sysfs(pci_adres)
+                self.log.info("from Sysfs:%s", sys_pci_id)
```
Comment (narasimhan-v) [inline io/pci/pci_info_lsvpd.py]: PCI ID from Sysfs
PCI ID from Vpd

**Comment by Naresh-ibm** [comment]
made the suggested changes . please have a look and let me know if you need any more info.

**Comment by PraveenPenguin** [comment]
@narasimhan-v thanks

**Pull request merged**